File: app/src/main/java/com/wzq/sample/ui/recycler_single_network/detail/DetailViewModel.java
package com.wzq.sample.ui.recycler_single_network.detail;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.databinding.ObservableField;

import com.wzq.mvvmsmart.base.BaseViewModelMVVM;
import com.wzq.sample.bean.DemoBean;

public class DetailViewModel extends BaseViewModelMVVM {
    public ObservableField<DemoBean.ItemsEntity> entity = new ObservableField<>();

    public DetailViewModel(@NonNull Application application) {
        super(application);
    }

    public void setDemoEntity(DemoBean.ItemsEntity entity) {
        this.entity.set(entity);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        entity = null;
    }
}

File: app/src/main/java/com/wzq/sample/ui/recycler_network/detail/DetailViewModel.kt
package com.wzq.sample.ui.recycler_network.detail

import android.app.Application
import androidx.databinding.ObservableField
import com.wzq.mvvmsmart.base.BaseViewModelMVVM
import com.wzq.sample.bean.DemoBean.ItemsEntity

class DetailViewModel(application: Application) : BaseViewModelMVVM(application) {
    var entity: ObservableField<ItemsEntity?>? = ObservableField()
    fun setDemoEntity(entity: ItemsEntity?) {
        this.entity!!.set(entity)
    }

    override fun onDestroy() {
        super.onDestroy()
        entity = null
    }
}